src/mgr/MgrClient.cc

```diff
@@ -317,14 +330,20 @@ void MgrClient::send_report()
   session->con->send_message(report);
 
   if (stats_period != 0) {
-    report_callback = new FunctionContext([this](int r){send_report();});
+    report_callback = new FunctionContext([this](int r){_send_report();});
```

Comment:
@tchaikov Please review this, registering report_callback func to _send_report instead of send_report.

Reply:
are you also backporting 5faf1787573bb4896a67889471b3c033dbeacf04 ? i don't really think it's necessary in this PR.

we can just introduce `_send_pgstats()` which sends message without holding the lock, and let `send_report()` call it instead.

Reply:
Ack ! So it's fine to not include MgrClient::_send_stats changes in this PR ?

Reply:
@pdvian yeah, i think so. the `MgrClient::_send_stats()` was just a refactor. but it's fine either way. probably it's better this way to alleviate the trouble of future backports. but if you and including 5faf1787573bb4896a67889471b3c033dbeacf04 in this backport, please just cherry-pick it, instead of including it as a conflict resolution.

Reply:
@tchaikov Ack! Let me re-work on it.